fdw/gpusim_fdw/__init__.py

```diff
@@ -18,18 +18,28 @@ def __init__(self, options, columns):
         server = options['server']
         port = options['port']
 
-        data = {'smiles': input_smiles, 'return_count': self.max_results}
-
-        endpoint = GPUSIM_ENDPOINT.format(server, port, input_db)
-        response = requests.post(endpoint, data)
-        if not response.ok:
-            raise RuntimeError("Server connection failed")
-        self.data = response.json()
+        self.query = {'smiles': input_smiles, 'return_count': self.max_results}
+        self.endpoint = GPUSIM_ENDPOINT.format(server, port, input_db)
+        self._default_smiles = input_smiles
+        self._last_query_smiles = None
 
     def execute(self, quals, columns):
+        smiles = self._default_smiles
+        for qual in quals:
+            if qual.field_name == 'query' and qual.operator == '=':
+                smiles = qual.value
+                break
+        if smiles != self._last_query_smiles:
```

Comment:
We may just want to move this completely out of table definition, and make you do it every time you do the search?  I don't love that there's two ways to define the search query.  It gives some slight performance optimization for re-searching, but I doubt it's even that much given how fast gpusim is...

Reply:
Makes sense to me